_posts/2024-07-03-oidc-cient-request.adoc

```diff
@@ -0,0 +1,467 @@
+The feature also includes the ability to sign and/or encrypt the JWT for added security. https://openid.net/developers/how-connect-works/[OpenID Connect] is an identity layer on top of the OAuth 2.0 protocol
+that makes it possible for a client to verify a user’s identity based on authentication that’s performed by an OpenID
+provider. The OAuth 2.0 request syntax sends the Request Object in the request by adding them directly to the URL.
+However, OpenID Connect allows the Request Object to be sent as a JWT using https://openid.net/specs/openid-connect-core-1_0.html#JWTRequests[request parameters], which can be signed and optionally
+encrypted. This adds an added layer of protection, as the parameters are no longer human-readable.
+
+toc::[]
+
+== Prerequisites 
+
+To follow along with this guide, you will need:
+
+* Roughly 20 minutes (not including the time needed for building)
+
+* Access to an OpenShift cluster (try the Red Hat Developer Sandbox for free)
+
+* https://docs.openshift.com/container-platform/4.15/cli_reference/openshift_cli/getting-started-cli.html[OpenShift CLI]
+
+* https://helm.sh/docs/intro/install/[Helm Chart]
+
+* https://github.com/wildfly-security-incubator/elytron-examples[Elytron-examples project]
+
+== Example Application
+
+We will use a simple web application in this guide that consists of a https://github.com/wildfly-security-incubator/elytron-examples/blob/main/elytron-oidc-request/src/main/java/org/wildfly/security/examples/SecuredServlet.java[single servlet]. We will secure this servlet using OIDC.
+
+We will use the example in the https://github.com/wildfly-security-incubator/elytron-examples/tree/main/elytron-oidc-request[elytron-oidc-request] directory in this repo.
+
+To obtain this example, clone the elytron-examples repository to your local machine:
+
+```
+git clone git@github.com:wildfly-security-incubator/elytron-examples.git
+```
+
+== Setting up the KeyPairs
+
+In order for us to sign the JWT's, we will need to use key pairs corresponding to the algorithms specified. These are tokens communicated between the client and the server and are used to sign and encrypt JWT's.
+
+=== About Key Pairs
+
+Typically key pairs have 3 components: 
+
+* *Private Key:* This is the part of the key that must be kept secret and will be used to sign the JWT. Your OpenID provider communicates the public key of some keyPair(s) to you. The public key is used by you to encrypt the JWT.
+
+* *Public Key:* This is the part of the keypair that is shared with trusted parties. Typically, the OpenID provider shared a set of public keys with us which we use to encrypt our JWT's among other things. Public keys can be extracted from certificates.
+
+* *Certificate:* This is used to verify the signatures among other things. We share the certificates of keypairs we generate with the OpenID provider, and it contains the public key.
```

Comment:
s/keypairs/key pairs (please update all references in this doc)

Reply:
fixed!